subsys/fs/nvs/nvs.c

```diff
@@ -418,10 +457,18 @@ static int nvs_flash_wrt_entry(struct nvs_fs *fs, uint16_t id, const void *data,
 
 	nvs_ate_crc8_update(&entry);
 
-	rc = nvs_flash_data_wrt(fs, data, len);
+	rc = nvs_flash_data_wrt(fs, data, len, true);
 	if (rc) {
 		return rc;
 	}
+
+#ifdef CONFIG_NVS_DATA_CRC
+	/* No CRC has been added if this is a deletion write request */
+	if (len > 0) {
+		entry.len += NVS_DATA_CRC_SIZE;
```

Comment:
we can move this up to avoid computing crc8 twice when CONFIG_NVS_DATA_CRC is enabled
you can do :
#ifdef CONFIG_NVS_DATA_CRC
if (len > 0) {
		entry.len = (uint16_t)len + NVS_DATA_CRC_SIZE;
}
else {
               entry.len = (uint16_t)len;
}

Reply:
Yes, it's better to compute the CRC only once.